src/oss/langchain/voice_agent.mdx

```diff
@@ -0,0 +1,764 @@
+
+This guide demonstrates how to interact with LangChain [agents](/oss/langchain/agents) through voice channels. It follows the reference architecture implemented in the LangChain repo [here](https://github.com/langchain-ai/voice-sandwich-demo).
+
+### Concepts
+
+There are two common architectures for voice agents:
+
+#### 1. "The Sandwich"
+
+The Sandwich architecture composes three distinct components: speech-to-text (STT), a text-based LangChain agent, and text-to-speech (TTS).
+
+```mermaid
+flowchart LR
+    A[User Audio] --> B[Speech-to-Text]
+    B --> C[LangChain Agent]
+    C --> D[Text-to-Speech]
+    D --> E[Audio Output]
+```
+
+**Pros:**
+- Full control over each component (swap STT/TTS providers as needed)
+- Access to latest capabilities from modern text-modality models
+- Transparent behavior with clear boundaries between components
+
+**Cons:**
+- Requires orchestrating multiple services
+- Additional complexity in managing the pipeline
+- Conversion from speech to text loses information (e.g., tone, emotion)
+
+#### 2. Speech-to-Speech Architecture
+
+Speech-to-speech uses a multimodal model that processes audio input and generates audio output natively.
+
+```mermaid
+flowchart LR
+    A[User Audio] --> B[Multimodal Model]
+    B --> C[Audio Output]
+```
+
+**Pros:**
+- Simpler architecture with fewer moving parts
+- Typically lower latency for simple interactions
+- Direct audio processing captures tone and other nuances of speech
+
+**Cons:**
+- Limited model options, greater risk of provider lock-in
+- Features may lag behind text-modality models
+- Less transparency in how audio is processed
+- Reduced controllability and customization options
+
+This guide demonstrates the **sandwich architecture** to balance performance, controllability, and access to modern model capabilities. The sandwich can achieve sub-700ms latency with some STT and TTS providers while maintaining control over modular components.
+
+### Demo Application Overview
+
+We will walk through building a voice-based agent using the sandwich architecture. The agent will manage orders for a sandwich shop. The application will demonstrate all three components of the sandwich architecture, using [AssemblyAI](https://www.assemblyai.com/) for STT and [ElevenLabs](https://elevenlabs.io/) for TTS (although adapters can be built for most providers).
+
+An end-to-end reference application is available in the [voice-sandwich-demo](https://github.com/langchain-ai/voice-sandwich-demo) repository. We will walk through that application here.
+
+The demo uses WebSockets for real-time bidirectional communication between the browser and server. The same architecture can be adapted for other transports like telephony systems (Twilio, Vonage) or WebRTC connections.
+
+## Setup
+
+For detailed installation instructions and setup, see the [repository README](https://github.com/langchain-ai/voice-sandwich-demo#readme).
+
+### Architecture Overview
```

Comment:
I think it makes more sense to put the Architecture overview under the Demo application overview section and retitle it to "Architecture"